src/hulls/graham.jl

```diff
@@ -43,13 +43,28 @@ function hull(points, ::GrahamScan)
   θ = [∠(A, O, B) for B in q]
   q = q[sortperm(θ)]
 
+  # initialise and skip collinear points at beginning 
+  r = [O]
+  idx = 1
+  while idx ≤ length(q) && coordinates(q[idx])[2] == coordinates(O)[2]
+    idx += 1
+  end
+  idx = max(idx, 2)
+  push!(r, q[idx - 1])
+  if idx > length(q)
+    return PolyArea(r)
+  else
+    push!(r, q[idx])
+  end
```

Comment:
It seems that this code can be simplified with the use of `findfirst`. You can find the first index satisfying a condition and then initialize `r`. If no index satisfies the condition, then the result is `nothing` and you can return the trivial `PolyArea`. Can you please adjust the code accordingly?

Reply:
How did I forget that `findfirst` exists... Yes, done.